### ISSUE TITLE
Lack of Guidance on Implementing Classical Controlled Instructions in Braket Using OpenQASM 3

### ISSUE THREAD

**Issue by contra-bit** [amazon-braket/Braket.jl#101]
**What did you find confusing? Please describe.**  
I tried to implement classical controlled instructions in Braket. However, I didn't understand how to properly create and utilize classical control structures within the Braket framework, as the documentation does not provide clear examples or guidance on this topic.

**Describe how documentation can be improved**  
The documentation could be improved by including specific examples of classical controlled instructions in Braket, particularly focusing on how to implement if-else statements similar to the OpenQASM 3 example provided. It would be helpful to have a dedicated section that explains the syntax and semantics of classical control structures, along with practical examples that demonstrate their usage in quantum circuits.

**Additional context**  
I am trying to achieve the following code in OpenQASM 3.0:

```qasm
OPENQASM 3.0;
bit[2] b;
qubit[3] q;
h q[1];
b[0] = measure q[1];
if (b[0]) {
  x q[0];
}
```

However, I am unsure how to translate this logic into the Braket environment. Any guidance or examples would be greatly appreciated.

This is related to this [issue](https://github.com/amazon-braket/BraketSimulator.jl/issues/47)

**Comment by speller26** [comment]
At the moment, we don't support classical controlled instructions; we do support them in [autoqasm](https://github.com/amazon-braket/autoqasm), which provides a Python-native syntax for expressing programs, but this currently an experimental library.

**Comment by rmshaffer** [comment]
To add to @speller26's response, [here's an example notebook](https://github.com/amazon-braket/autoqasm/blob/main/examples/2_Expressing_classical_control_flow.ipynb) showing how this would be done in AutoQASM.

**Comment by Lung-Yi** [comment]
@rmshaffer Hi sir, I want to add a question, does the real quantum device (any backend) on AWS support the classical control flow by AutoQASM? Thanks!

**Comment by rmshaffer** [comment]
@Lung-Yi No, the devices currently available through Braket do not support classical control flow based on measurement results.
